### PR TITLE
2.32.0 release

### PR DESCRIPTION
**Pull request by matthewfala**
### 2.32.0 Release
This release includes:
* Fluent Bit [1.9.10](https://github.com/fluent/fluent-bit/tree/v1.9.10)
* Amazon CloudWatch Logs for Fluent Bit 1.9.4
* Amazon Kinesis Streams for Fluent Bit 1.10.2
* Amazon Kinesis Firehose for Fluent Bit 1.7.2

Compared to `2.31.12.20230911` this release adds:
* Feature - Amazon Managed Prometheus support added via sigv4 authentication on `prometheus_remote_write` [aws-for-fluent-bit:256](https://github.com/aws/aws-for-fluent-bit/issues/256)
* Enhancement - Allow multiple `cloudwatch_log` output workers by removing sequence tokens from API requests [aws-for-fluent-bit:526](https://github.com/aws/aws-for-fluent-bit/issues/526)
* Bug - Fix multiline input behavior when multiple streams are parsed (stderr, stdout) together. Multiline logs are no longer terminated when streams are switched between [fluent-bit:7469](https://github.com/fluent/fluent-bit/pull/7469)

### CODE REVIEW

**Comment by matthewfala** [comment]
Please see the following cherry-pick diffs:
Cloudwatch Sequence Token Deprecation: https://github.com/matthewfala/fluent-bit/pull/58
Prometheus: https://github.com/matthewfala/fluent-bit/pull/59
Multiline flush fix: https://github.com/matthewfala/fluent-bit/pull/60

**Comment by matthewfala** [comment]
Also updated the cherry-picking system in the docker build file to end build on a single failure.

**Comment by matthewfala** [comment]
Updated the master and CWL cherry-pick: https://github.com/fluent/fluent-bit/compare/6c9656d3d7e965faf7ee0fdbe657f2775bc58c24..737cb822fca7274d76c2e15c2ef6ff6faae872d0

**Comment by matthewfala** [comment]
Updated

File: AWS_FLB_CHERRY_PICKS [hunk 1]
```diff
@@ -83,15 +84,6 @@ https://github.com/fluent/fluent-bit.git 1.9 81cdf7eced4e420043277237fba092157b1
...
 
-# TCP input: user friendly message for skipping logs https://github.com/fluent/fluent-bit/pull/6061
-https://github.com/fluent/fluent-bit.git 1.9 c5d57ccf4f85af77e2a2bbb1121df95887e2962f
-
-# ElasticSearch: fix buffer bulk over run https://github.com/fluent/fluent-bit/pull/5770
-https://github.com/fluent/fluent-bit.git 1.9 d3a0697c3c4a876250cc6c2c4ff44cd77908ea3c
-
-# windows: fix time zone offset https://github.com/fluent/fluent-bit/pull/6368
-https://github.com/fluent/fluent-bit.git 1.9 cae9a8f9cb6eab7449f14d914f9ba1d8d4e93cf0
-
```
Comment (PettitWesley) [inline AWS_FLB_CHERRY_PICKS:94]: @matthewfala why are these deleted?

Comment (matthewfala) [reply AWS_FLB_CHERRY_PICKS:94]: All of these PRs were merged int 1.9 series by v1.9.10. Please see:
https://github.com/fluent/fluent-bit/pull/6368
https://github.com/fluent/fluent-bit/pull/5770
https://github.com/fluent/fluent-bit/pull/6061

You can see all of them in the commit history here: https://github.com/fluent/fluent-bit/commits/v1.9.10?before=760956f50cdc7eef3047df20d2299202a8c68594+35&branch=v1.9.10&qualified_name=refs%2Ftags%2Fv1.9.10

**Comment by matthewfala** [comment]
Update wording to
Compared to 2.31.12.20230911 this release adds:

    - Feature - Customers can send metrics to Amazon Managed Prometheus via added sigv4 authentication on prometheus_remote_write https://github.com/aws/aws-for-fluent-bit/issues/256
    - Enhancement - Customers can route logs to CloudWatch Logs at higher throughputs by increasing number of output workers as cloudwatch output plugin removed sequence tokens from API requests https://github.com/aws/aws-for-fluent-bit/issues/526
    - Fix - Fix multiline input behavior when multiple streams are parsed (stderr, stdout) together. Multiline logs are no longer terminated when streams are switched between https://github.com/fluent/fluent-bit/pull/7469

- Update public doc - Request for doc change (not-blocking)
- Update public doc - AWS FireLens Example (9/28)
    - Add link to release notes
- Update public docs once

File: AWS_FLB_CHERRY_PICKS (hunk 1 above)
Comment (PettitWesley) [reply AWS_FLB_CHERRY_PICKS:94]: ah, thanks

**Review by PettitWesley — APPROVED** [review]

**Comment by matthewfala** [comment]
Update the examples: https://github.com/aws-samples/amazon-ecs-firelens-examples/pull/125
Update cloudwatch_logs guidance issue: https://github.com/aws/aws-for-fluent-bit/issues/340

**Comment by matthewfala** [comment]
Updated cloudwatch_logs cherrypick https://github.com/matthewfala/fluent-bit/compare/4c92b10638b02cd6203219464a24dc7ad54c1acb..8ee560e388bbbf850069c81bbca06275f330baeb

https://github.com/aws/aws-for-fluent-bit/compare/133a592172c350e9711cf6ab1232b2de14b8c2a8..44ac0a9aa454aaa16f0e3fe796366551b1a3a67e

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
+### 2.32.0
+This release includes:
+* Fluent Bit [1.9.10](https://github.com/fluent/fluent-bit/tree/v1.9.10)
+* Amazon CloudWatch Logs for Fluent Bit 1.9.4
+* Amazon Kinesis Streams for Fluent Bit 1.10.2
+* Amazon Kinesis Firehose for Fluent Bit 1.7.2
+
+Compared to `2.31.12.20230911` this release adds:
+* Feature - Amazon Managed Prometheus support added via sigv4 authentication on `prometheus_remote_write` [aws-for-fluent-bit:256](https://github.com/aws/aws-for-fluent-bit/issues/256)
+* Enhancement - Allow multiple `cloudwatch_log` output workers by removing sequence tokens from API requests [aws-for-fluent-bit:526](https://github.com/aws/aws-for-fluent-bit/issues/526)
```
Comment (PettitWesley) [inline CHANGELOG.md]: Let's link to the new firelens example somewhere in the release notes.

Comment (PettitWesley) [reply CHANGELOG.md]: the final link is predictable

File: Makefile [hunk 3]
```diff
@@ -28,7 +28,7 @@ release: build build-init
 debug: main-debug init-debug
 
 .PHONY: build
-build: linux-plugins
```
Comment (PettitWesley) [inline Makefile:31]: build should still build linux-plugins?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,17 @@
 # Changelog
 
+### 2.32.0
+This release includes:
+* Fluent Bit [1.9.10](https://github.com/fluent/fluent-bit/tree/v1.9.10)
+* Amazon CloudWatch Logs for Fluent Bit 1.9.4
+* Amazon Kinesis Streams for Fluent Bit 1.10.2
+* Amazon Kinesis Firehose for Fluent Bit 1.7.2
+
+Compared to `2.31.12.20230911` this release adds:
```
Comment (matthewfala) [inline CHANGELOG.md]: These notes are not updated. Resolve before merge

Comment (PettitWesley) [reply CHANGELOG.md]: Also the release note should be Compared to `2.31.12`, the date stamped re-builds for linux only don't count as a full release.

File: Makefile (hunk 3 above)
Comment (matthewfala) [reply Makefile:31]: Build dockerfile/image does not depend on the linux-plugins image. Actually only the release and main-release depends on the "linux plugins"

It was slowing down my fluent-bit build tests.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,17 @@
...
 
+### 2.32.0
+This release includes:
+* Fluent Bit [1.9.10](https://github.com/fluent/fluent-bit/tree/v1.9.10)
+* Amazon CloudWatch Logs for Fluent Bit 1.9.4
+* Amazon Kinesis Streams for Fluent Bit 1.10.2
+* Amazon Kinesis Firehose for Fluent Bit 1.7.2
+
+Compared to `2.31.12` this release adds:
+* Feature - Customers can send metrics to Amazon Managed Prometheus via added sigv4 authentication on `prometheus_remote_write`. Refer to [amazon-ecs-firelens-examples](https://github.com/aws-samples/amazon-ecs-firelens-examples/blob/mainline/examples/fluent-bit/amazon-managed-service-for-prometheus/README.md) for information on how to export metrics to AMP on ECS [aws-for-fluent-bit:256](https://github.com/aws/aws-for-fluent-bit/issues/256)
```
Comment (matthewfala) [inline CHANGELOG.md:11]: Added link to prometheus example

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,17 @@
 # Changelog
 
+### 2.32.0
+This release includes:
+* Fluent Bit [1.9.10](https://github.com/fluent/fluent-bit/tree/v1.9.10)
+* Amazon CloudWatch Logs for Fluent Bit 1.9.4
+* Amazon Kinesis Streams for Fluent Bit 1.10.2
+* Amazon Kinesis Firehose for Fluent Bit 1.7.2
+
+Compared to `2.31.12` this release adds:
```
Comment (matthewfala) [inline CHANGELOG.md:10]: Updated version comparison to 2.31.12

File: CHANGELOG.md (hunk 4 above)
Comment (matthewfala) [reply CHANGELOG.md]: Sure! Updated.

File: CHANGELOG.md (hunk 2 above)
Comment (matthewfala) [reply CHANGELOG.md]: Sure! Link added.

**Comment by matthewfala** [comment]
Consider adding the following patches:
- Coroutine drop fix: https://github.com/fluent/fluent-bit/pull/7728
- Throttle fix: https://github.com/matthewfala/fluent-bit/commit/7b05b7ebfe55261ed12d5006c8b682572b6abf4c

**Comment by matthewfala** [comment]
Throttle fix is a backport of: https://github.com/fluent/fluent-bit/pull/6500/files

**Comment by matthewfala** [comment]
Created 1.9 backport of the network fix: https://github.com/matthewfala/fluent-bit/tree/2.32.0-premature-connection-destruction

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,20 @@
...
+* Feature - Support multiline parsers with the init tag. Multiline parsers can be specified same as a standard parser file [aws-for-fluent-bit:537](https://github.com/aws/aws-for-fluent-bit/issues/537)
+* Enhancement - Customers can route logs to CloudWatch Logs at higher throughputs by increasing number of output workers as `cloudwatch_logs` output plugin removed sequence tokens from API requests [aws-for-fluent-bit:526](https://github.com/aws/aws-for-fluent-bit/issues/526)
+* Fix - Fix multiline input behavior when multiple streams are parsed (stderr, stdout) together. Multiline logs are no longer terminated when streams are switched between [fluent-bit:7469](https://github.com/fluent/fluent-bit/pull/7469)
+* Fix - Fix networking edgecase causing data loss and OOM related issues via net timeout event injection resolution [fluent-bit:7728](https://github.com/fluent/fluent-bit/pull/7728/files)
+* Fix - Allow reduced filter_throttle log output with `print_status` `false` configuration option fix [fluent-bit:7469](https://github.com/fluent/fluent-bit/pull/6500/files)
+
```
Comment (PettitWesley) [inline CHANGELOG.md:17]: @matthewfala we should add on3e more things from here, when I tried to stage 2.32: https://github.com/aws/aws-for-fluent-bit/pull/658/files

1. AWS Profile support

Comment (matthewfala) [reply CHANGELOG.md:17]: Documenting our decision:
1. Will further review this feature item in the future in the 2.33.0 release. Requires further review and testing. [link](https://github.com/fluent/fluent-bit/pull/7127)
2. Proceed with release

**Comment by matthewfala** [comment]
For documentation: Test the network fix 2.32.0-premature-connection-destruction change via:
```
diff --git a/src/flb_network.c b/src/flb_network.c
index f8cc7068d..d63d7ab94 100644
--- a/src/flb_network.c
+++ b/src/flb_network.c
@@ -441,7 +441,7 @@ static int net_connect_async(int fd,
     ret = mk_event_add(u_conn->evl,
                        fd,
                        FLB_ENGINE_EV_THREAD,
-                       MK_EVENT_WRITE, &u_conn->event);
+                       MK_EVENT_READ, &u_conn->event);
     u_conn->event.priority = FLB_ENGINE_PRIORITY_CONNECT;
     if (ret == -1) {
         /*
```

This will cause the network events to artificially timeout, causing the injection timeout code which is modified to be triggered.

**Comment by matthewfala** [comment]
Add update to the error log to include api caller: https://github.com/fluent/fluent-bit/pull/8122/files
